packages/analytics-js-common/src/v1.1/utils/storage/cookie.js

```diff
@@ -24,13 +24,17 @@ class CookieLocal {
     let domain = `.${topDomain(window.location.href)}`;
     if (domain === '.') domain = null;
 
-    // the default maxage and path
-    this.cOpts = defaults(inOpts, {
-      maxage: 31536000000,
-      path: '/',
-      domain,
-      samesite: 'Lax',
-    });
+    // Override the default options with the provided options
+    this.cOpts = mergeDeepRight(
```

Comment:
MergeDeepRight needs some comments to explain its functionality and some tests to verify it. The tests will also help better clarify the function's usage.

Reply:
Done. Please see, https://github.com/rudderlabs/rudder-sdk-js/pull/2046/commits/ad8bfc88d48c3aca537497869550c3be23348d21